src/components/SubcategoryTable.js

```diff
@@ -0,0 +1,123 @@
+import ProgressBar from '@ramonak/react-progress-bar';
+import { Icon } from '@iconify/react';
+import { sectionColors } from '../lib/coverageSectionIds';
+import { getProfileFieldsCoveredCount } from '../lib/coverageStats/statsUtils';
+
+function SubcategoryTable({ className, selectedSection, coverageData }) {
+  let sectionData;
+  if (selectedSection === 'Overall') {
+    sectionData = [];
+    coverageData.forEach((section) => {
+      sectionData.push(...section.data);
+    });
+  } else {
+    sectionData = coverageData.filter((x) => x.section === selectedSection)[0]?.data;
+  }
+  const profiles = sectionData?.map((profile) => {
+    const fields = getProfileFieldsCoveredCount(profile, selectedSection);
+    return {
+      name: profile.profile,
+      covered: fields.map((f) => f.covered).reduce((a, b) => a + b, 0),
+      total: fields.map((f) => f.total).reduce((a, b) => a + b, 0),
+      fields,
+    };
+  });
+
+  return (
+    <div className={`${className} flex flex-col bg-white m-2 rounded-widgit w-1/2 h-96`}>
+      {/* Header */}
+      <h3 className="p-2 font-sans font-semibold text-xl">
+        <span className={`${sectionColors.text[selectedSection]}`}>{selectedSection}</span> Subcategories
+      </h3>
+      <div className="grow overflow-y-auto">
+        {/* Table of profiles */}
+        <table className="w-full table-fixed text-left">
+          <thead className="bg-gray-100 text-gray-500 font-thin rounded">
+            <tr>
+              <th scope="col" className="px-6 font-normal py-3">
+                Name
+              </th>
+              <th scope="col" className="font-normal py-3">
+                Coverage
+              </th>
+              <th scope="col" className="font-normal py-3">
+                Link
+              </th>
+            </tr>
+          </thead>
+          <tbody>
+            {profiles?.map((profile) => (
+              <tr className="h-10 border-y-2 border-gray-200" key={profile.name}>
+                <td className="h-10 hover:bg-gray-200" colSpan={3}>
+                  <details>
```

Comment:
I feel like instead of using the details element here we might just want to use state to determine which table rows should be expanded and hide the un-expanded ones accordingly. We're really not using the parent table class correctly here, and we probably shouldn't be nesting tables in the section below. We also should probably have a class applied conditionally when the sub-table is open/closed that adds a `:before` psuedoelement to the first table cell with the arrow pointing to the side or down.

Reply:
I used the summary/detail stuff because it seemed like a convenient way to avoid storing open/closed state, but then it ended up being a nightmare to style, so I'm fine changing it (you can't easily style the little arrow thingy it adds with Tailwind, which is why I ended up removing it with CSS)

Reply:
I figured, it just ended up being tough because we're nested within a table and everything. Sorry it didn't work out this way!